src/Hosting/SAF.Hosting/ServiceProviderExtensions.cs

```diff
@@ -12,12 +14,38 @@ public static class ServiceProviderExtensions
     {
         public static IServiceProvider UseServiceHost(this IServiceProvider serviceProvider)
         {
+            ILogger logger = serviceProvider.GetService<ILogger<ServiceHost>>();
+            int testCount = 10;
+            while (!serviceProvider.IsConnected())
```

Comment:
The idea was to ensure that the SAF plug-in services will be loaded after the infrastructure has fully been initialized. First this code only considers the `IStorageInfrastructure` and not the `IMessagingInfrastructure`. In addition each infrastructure implementation should handle the initial connect attempts and initializations itself.

If possible `UseServiceHost` should not "ask" any registrered infrastructure service implementation whether its connected or not.